source/Pages/jupyter_notebooks.rst

```diff
@@ -7,6 +8,147 @@ Jupyter Notebooks
+
+First we start by fetching a container:
+
+.. code-block:: bash
+
+   singularity build jupyter.sif docker://jupyter/scipy-notebook:latest
+
+This will pull one of the official jupyter containers from docker hub, and build a singularity container from it. This container encapsulates the entire environment and can be entered to start a notebook session. More official containers can be found `here <https://hub.docker.com/r/jupyter/>`_, and more docker images in general can be found at `docker hub <https://hub.docker.com/>`_.
+
+After the build procedure is complete, you can start the jupyter instance with 
+
+.. code-block:: bash
+
+   singularity run jupyter.sif
+
+which will automatically start the instance. Alternatively, you can start an interactive shell session in the container and start it manually:
+
+.. code-block:: bash
+
+   singularity shell jupyter.sif
+   jupyter lab
+
+To receive the notebook locally in your browser, as mentioned above, a tunnel has to be opened in a new terminal, with:
+
+.. code-block:: bash
+   
+   ssh -NL 8888:wn-db-01:8888 spider
+
+Where, again, the machine name and port name have to match where you are running the job and the port chosen by jupyter, respectively. Now you can open the link provided by jupyter, which has the shape of ``http://localhost:8888/lab?token=abc123``.
+
+If the forwarding or other steps do not work, please contact :ref:`our helpdesk <helpdesk>`.
+
+Custom image
+============
+
+Images can be customised to suit your needs, by adding extra steps during the build process. This is done with so-called 'definitions' files. These are plaintext files with instructions for the singularity build. For a full overview, see the `singularity documentation <https://docs.sylabs.io/guides/latest/user-guide/definition_files.html>`_. Here is a small example that can be expanded. Make a file called ``jup-custom.def`` and fill it with:
+
+.. code-block:: bash
+
+   Bootstrap: docker
+   From: jupyter/scipy-notebook:latest
+
+   %post
+     pip install docopt
+
+   %runscript
+     jupyter lab --ip=0.0.0.0
+
+   %help
+     This is a demo container to show how to run jupyter lab 
+
+You can build this with:
+
+.. code-block:: bash
+
+   singularity build jup-custom.sif jup-custom.def
+
+and once it is finished building, you can enter the `sif` file with the ``singularity shell`` command, or start jupyter directly with ``singularity run``. You still have to forward the connection as described above before you can open the notebook in a browser.
+
+To get a full overview of what is possible during building in terms of installing packages, raising permissions, setting paths, mounting local folders and more, see the `official singularity documentation <https://docs.sylabs.io/guides/latest/user-guide/definition_files.html>`_.
+
```

Comment:
Consider adding a note on how to save the notebook ('Save as' will store it by default in `$HOME`).

Reply:
I'm adding a link to the full jupyter docs, that seems more appropriate (as there is none in the whole document).